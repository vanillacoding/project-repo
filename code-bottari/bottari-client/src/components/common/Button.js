import PropTypes from "prop-types";
import styled, { css } from "styled-components";

const buildStyle = ({
  width,
  padding,
  margin,
  border,
  borderRadius,
  backgroundColor,
  color,
  fontSize,
  whiteSpace,
  boxShadow,
  hoverBackgroundColor,
  hoverColor,
  hoverBoxShadow,
  activeBackgroundColor,
  activeBoxShadow,
  focusBackgroundColor,
  focusColor,
}) => css`
  width: ${width};
  padding: ${padding};
  margin: ${margin};
  border: ${border};
  border-radius: ${borderRadius};
  background-color: ${backgroundColor};
  color: ${color};
  font-size: ${fontSize};
  white-space: ${whiteSpace};
  box-shadow: ${boxShadow};

  &:hover {
    background-color: ${hoverBackgroundColor};
    color: ${hoverColor};
    box-shadow: ${hoverBoxShadow};
  }

  &:active {
    background-color: ${activeBackgroundColor};
    box-shadow: ${activeBoxShadow};
  }

  &:focus {
    background-color: ${focusBackgroundColor};
    color: ${focusColor};
  }
`;

const buttonType = {
  basic: {
    hoverBackgroundColor: "#26BFA6",
    activeBackgroundColor: "#21A18C",
  },
  userMenu: {
    width: "180px",
    margin: "2px 0px",
    backgroundColor: "#AEA5DF",
    boxShadow: "0px 2px 4px rgba(0, 0, 0, 0.25)",
    hoverBackgroundColor: "#26BFA6",
    activeBackgroundColor: "#21A18C",
  },
  search: {
    padding: "12px 38px",
    backgroundColor: "#8877E8",
    fontSize: "20px",
    hoverBackgroundColor: "#26BFA6",
    activeBackgroundColor: "#21A18C",
  },
  home: {
    padding: "20px 25px",
    borderRadius: "5px",
    backgroundColor: "#9871B7",
    hoverBackgroundColor: "#26BFA6",
    activeBackgroundColor: "#21A18C",
  },
  edit: {
    padding: "15px 20px",
    border: "1px solid #C6CECD",
    borderRadius: "5px",
    backgroundColor: "#FFFFFF",
    color: "#9871B7",
    boxShadow: "1px 1px 4px rgba(0, 0, 0, 0.25)",
    hoverBackgroundColor: "#9871B7",
    hoverColor: "#FFFFFF",
    activeBackgroundColor: "#26BFA6",
  },
  filter: {
    border: "1px solid #C6CECD",
    borderRadius: "5px",
    backgroundColor: "#FFFFFF",
    color: "#9871B7",
    fontSize: "20px",
    boxShadow: "1px 1px 5px rgba(0, 0, 0, 0.25)",
    focusBackgroundColor: "#9871B7",
    focusColor: "#FFFFFF",
  },
  notification: {
    padding: "10px 10px",
    width: "430px",
    margin: "2px",
    border: "1px solid #AEA5DF",
    backgroundColor: "#FFFFFF",
    color: "#000000",
    fontSize: "14px",
    boxShadow: "0px 1px 3px 1px rgba(0, 0, 0, 0.25)",
    hoverBoxShadow: "0px 2px 5px 3px rgba(0, 0, 0, 0.25)",
    activeBackgroundColor: "#DCF1EC",
  },
  tool: {
    padding: "20px 20px",
    margin: "1px",
    borderRadius: "3px",
    backgroundColor: "#FFFFFF",
    color: "#000000",
    whiteSpace: "nowrap",
    hoverBoxShadow: "0px 2px 3px 1px rgba(0, 0, 0, 0.25)",
    activeBoxShadow: "inset 0px 1px 1px 1px rgba(0, 0, 0, 0.25)",
  },
  utility: {
    padding: "10px 30px",
    margin: "2px",
    borderRadius: "5px",
    backgroundColor: "#FFFFFF",
    color: "#9871B7",
    fontSize: "18px",
    boxShadow: "1px 1px 4px rgba(0, 0, 0, 0.25)",
    hoverBoxShadow: "0px 2px 3px 1px rgba(0, 0, 0, 0.25)",
    activeBoxShadow: "inset 0px 1px 1px 1px rgba(0, 0, 0, 0.2)",
  },
  icon: {
    padding: "0px",
    backgroundColor: "transparent",
  },
  commentOption: {
    width: "90px",
    margin: "2px 0px",
    backgroundColor: "#AEA5DF",
    fontSize: "15px",
    boxShadow: "0px 2px 4px rgba(0, 0, 0, 0.25)",
    hoverBackgroundColor: "#26BFA6",
    activeBackgroundColor: "#21A18C",
  },
};

const StyledButton = styled.button`
  padding: 13px 30px;
  border: none;
  border-radius: 10px;
  background-color: #AEA5DF;
  color: #FFFFFF;
  font-size: 18px;
  line-height: 0.9;
  transition: 0.2s;

  ${({ variant }) => buildStyle(buttonType[variant])}
`;

export default function Button({
  type,
  variant,
  onClick,
  children,
  className,
}) {
  return (
    <StyledButton
      type={type}
      variant={variant}
      onClick={onClick}
      className={className}
    >
      {children}
    </StyledButton>
  );
}

Button.propTypes = {
  type: PropTypes.string,
  variant: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  className: PropTypes.string,
};

Button.defaultProps = {
  type: "button",
};
