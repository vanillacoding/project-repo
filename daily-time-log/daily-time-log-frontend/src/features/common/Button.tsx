/* eslint-disable @typescript-eslint/no-empty-function */
import React, { MouseEventHandler } from "react";
import styled, { css } from "styled-components";
import { boxShadow as boxShadowCss } from "../../assets/styles/utilsStyled";

interface Sizes {
  large: {
    height: string;
    fontSize: string;
  };
  medium: {
    height: string;
    fontSize: string;
  };
  small: {
    height: string;
    fontSize: string;
  };
}

interface Props {
  children: React.ReactNode;
  color?: "blue" | "pink";
  size: keyof Sizes;
  fullWidth?: boolean;
  background?: boolean;
  boxShadow?: boolean;
  onClick?: MouseEventHandler;
  disabled?: boolean;
}

function Button({
  children,
  color,
  size,
  fullWidth,
  background,
  boxShadow,
  disabled,
  onClick,
}: Props) {
  return (
    <ButtonStyled
      color={color}
      size={size}
      fullWidth={fullWidth}
      background={background}
      boxShadow={boxShadow}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </ButtonStyled>
  );
}

const boxShadowStyles = css<Props>`
  ${({ boxShadow }) => {
    return boxShadow && boxShadowCss;
  }};
`;

const fullWidthStyles = css<Props>`
  ${({ fullWidth }) => {
    return (
      fullWidth &&
      css`
        width: 100%;
        justify-content: center;
      `
    );
  }}
`;

const colorStyles = css<Props>`
  ${({ theme, color, background, disabled }) => {
    const selected = theme.palette[color];

    if (!background) {
      return css`
        border: none;
        background: none;
      `;
    }

    if (disabled) {
      return;
    }

    return css`
      background: ${selected};
      color: ${theme.color.buttonFont};

      :hover {
        background: ${theme.palette[`dark${color}`]};
      }
    `;
  }}
`;

const sizes: Sizes = {
  large: { height: "3rem", fontSize: "1.25rem" },
  medium: { height: "2.25rem", fontSize: "1rem" },
  small: { height: "1.75rem", fontSize: "0.875rem" },
};

const sizeStyles = css<Props>`
  ${({ size }) => css`
    height: ${sizes[size].height};
    font-size: ${sizes[size].fontSize};
  `}
`;

const ButtonStyled = styled.button`
  display: inline-flex;
  align-items: center;
  outline: none;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  padding-left: 1rem;
  padding-right: 1rem;

  ${colorStyles}
  ${sizeStyles}
  ${fullWidthStyles}
  ${boxShadowStyles}
`;

Button.defaultProps = {
  onClick: () => {},
  color: "blue",
  background: true,
  fullWidth: false,
  boxShadow: false,
  disabled: false,
};

export default Button;
