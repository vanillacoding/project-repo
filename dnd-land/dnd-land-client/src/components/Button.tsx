import React from "react";
import styled from "styled-components";

type ButtonProps = {
  text?: string;
  className?: string;
  children?: JSX.Element | JSX.Element[] | string;
  onClick: () => void;
};

function Button({ children, className, onClick, text }: ButtonProps) {
  return (
    <DefaultButton className={className} onClick={onClick}>
      {children}
      {text}
    </DefaultButton>
  );
}

const DefaultButton = styled.button`
  cursor: pointer;
  outline: none;
`;

export default Button;
