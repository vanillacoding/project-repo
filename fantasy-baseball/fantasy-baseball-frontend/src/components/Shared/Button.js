import React from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

import { selectColor, selectSize } from "../../styles/utils";

const Button = styled.button`
  padding: ${(props) => selectSize(props)};
  background: transparent;
  border: 1px solid ${(props) => selectColor(props)};
  font-family: "Bebas Neue";
  font-size: ${({ theme }) => theme.fontSize.middle};
  letter-spacing: 0.1rem;
  color: ${(props) => selectColor(props)};
  cursor: pointer;
  transition: 0.3s all;

  &:hover {
    background: ${({ theme }) => theme.color.blue};
    border: 1px solid ${({ theme }) => theme.color.blue};
    color: ${({ theme }) => theme.color.white};

    & > span {
      width: 70px;

      span {
        background: ${({ theme }) => theme.color.white};
      }
    }
  }

  &:disabled {
    cursor: not-allowed;
    opacity: 0.3;

    &:hover {
      background: inherit;
      border-color: inherit;
    }
  }
`;

const Arrow = styled.span`
  width: 40px;
  padding: 0 0 0 10px;
  display: inline-block;
  position: relative;
  transition: 0.3s all;
`;

const Line = styled.span`
  width: 100%;
  height: 1px;
  background: ${(props) => selectColor(props)};
  display: inline-block;
`;

const End = styled.span`
  width: 15px;
  height: 1px;
  background: ${(props) => selectColor(props)};
  display: inline-block;
  position: absolute;
  bottom: 8px;
  transform: rotate(-135deg);
  transform-origin: bottom left;
`;

function SharedButton(props) {
  const {
    type,
    title,
    color,
    size,
    handleClick,
    hasArrow,
    disabled,
  } = props;

  return (
    <Button
      type={type}
      color={color}
      size={size}
      onClick={handleClick}
      disabled={disabled}
    >
      {title}
      {hasArrow
        && (
          <Arrow>
            <Line color={color} />
            <End color={color} />
          </Arrow>
        )}
    </Button>
  );
}

SharedButton.propTypes = {
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  hasArrow: PropTypes.bool,
  disabled: PropTypes.bool,
};

SharedButton.defaultProps = {
  hasArrow: true,
  disabled: false,
};

export default SharedButton;
