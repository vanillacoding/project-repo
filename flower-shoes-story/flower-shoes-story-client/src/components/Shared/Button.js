import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Button = ({ type, children, onClick, className }) => {
  return (
    <DefaultButton type={type} className={className} onClick={onClick}>
      {children}
    </DefaultButton>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func,
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
};

Button.defaultProps = {
  type: "submit",
};

const DefaultButton = styled.button`
  display: inline-block;
  border: 2px solid #1a1a1e;
  padding: 10px 20px;
  font-size: 11px;
  font-weight: 600;
  text-transform: uppercase;
  letter-spacing: .1em;
  font-family: "adrianna-extended";
  color: #222;
`;

export default Button;
