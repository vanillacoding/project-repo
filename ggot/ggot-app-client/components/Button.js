import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity
} from 'react-native';

export default function Button({ title, onChange }) {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.button}
        onPress={() => onChange()}
      >
        <Text style={styles.text}>
          {title}
        </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F2F2F0',
    alignItems: 'center',
    justifyContent: 'center'
  },
  button: {
    backgroundColor: '#F2F2F0',
    borderWidth: 1,
    borderColor: '#BF0436',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 25,
    width: '80%',
    padding: 15
  },
  text: {
    fontSize: 12,
    color: '#474F59'
  }
});
