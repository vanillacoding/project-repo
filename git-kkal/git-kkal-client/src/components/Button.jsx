import styled from 'styled-components';

const Button = styled.button`
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid #ffffff43;
  cursor: pointer;
  background-color: #ffffff5a;
  color: white;
  text-align: center;
  text-decoration: none;

  &:hover {
    opacity: 0.6;
  }

  &:active {
    transform: translateY(4px);
  }
`;

export default Button;
