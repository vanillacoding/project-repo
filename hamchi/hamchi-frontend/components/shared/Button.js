import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import colors from '../../theme/color';

const Button = ({ text, type, onPress, customButtonStyle }) => {
  const btnBgColor = type === 'filled' ? colors.main : colors.transparent;
  const btnTextColor = type === 'filled' ? colors.white : colors.main;

  const buttonCommonStyle = {
    alignItems: "center",
    margin: 8,
    padding: 15,
    marginTop: 30,
    borderRadius: 30,
    backgroundColor: btnBgColor,
  };
  const textCommonStyle = {
    fontSize: 15,
    fontWeight: 'bold',
    color: btnTextColor,
  };

  return (
    <TouchableOpacity
      style={[buttonCommonStyle, customButtonStyle]}
      onPress={onPress}
    >
      <Text style={[textCommonStyle]}>{text}</Text>
    </TouchableOpacity>
  );
};

export default Button;
