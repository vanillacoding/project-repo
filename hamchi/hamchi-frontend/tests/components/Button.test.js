import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '../../components/shared/Button';

describe('Button component test', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress prop when pressed', () => {
    const handlePress = jest.fn();
    const buttonText = 'button';
    const { getByText } = render(
      <Button
        onPress={handlePress}
        text={buttonText}
      />
    );
    const button = getByText(buttonText);

    fireEvent.press(button);
    expect(handlePress).toHaveBeenCalledTimes(1);
  });
});
