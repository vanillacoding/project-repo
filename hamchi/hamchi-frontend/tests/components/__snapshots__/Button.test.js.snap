// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 30,
      "margin": 8,
      "marginTop": 30,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#F48D3B",
          "fontSize": 15,
          "fontWeight": "bold",
        },
      ]
    }
  />
</View>
`;
